app/models/user.rb

```diff
@@ -7,6 +7,7 @@ class User < ApplicationRecord
          :recoverable, :rememberable, :trackable, :validatable
 
   belongs_to :draw
+  belongs_to :group
```

Comment:
I don't know if Rails will know how to assign the `group_id` for the `leader` unless they're included in the `members` array, since this `belongs_to` corresponds to the `has_many :members` in the Group model, whereas a `has_one :group` will correspond to the `belongs_to :leader` in the Group model. Now, if we include the leader as a member (probably through a callback or something), then that will provide the `leader.group` method and we shouldn't need to use `has_one :group` here, and if we did include it anyway I'm not sure if it would break anything.

Does that make sense?